src/jsii-diagnostic.ts

```diff
@@ -628,6 +628,13 @@ export class JsiiDiagnostic implements ts.Diagnostic {
     name: 'language-compatibility/static-member-name-conflicts-with-nested-type',
   });
 
+  public static readonly JSII_5021_ABSTRACT_CLASS_MISSING_PROP_IMPL = Code.error({
+    code: 5021,
+    formatter: (intf: spec.InterfaceType, cls: spec.ClassType, prop: string) =>
+      `A declaration of "${intf.name}.${prop}" is missing on class "${cls.name}". Declare it as "public abstract" if the omission was on purpose.`,
```

Comment:
Also not wild about the part where it basically says: "If you wanted to omit this, you have to declare it". But I don't have a better suggestion. I guess you're trying to say "this was probably an unintentional mistake, but if you wanted to do this - though luck".

```suggestion
      `A declaration of "${intf.name}.${prop}" is missing on class "${cls.name}". Declare the property as "public abstract" if the omission was on purpose.`,
```